lib/exercism/metrics.rb

```diff
@@ -0,0 +1,45 @@
+require 'benchmark'
+require 'socket'
+
+class Metrics
+  VALID_METRIC_NAME = /\A[a-z_.]+\z/i
+
+  attr_reader :api_key, :host, :port
+
+  def self.time(metric_name, &block)
+    new.time(metric_name, &block)
+  end
+
+  def self.increment(metric_name)
+    new.increment(metric_name)
+  end
+
+  def initialize(api_key: ENV['HOSTEDGRAPHITE_APIKEY'], host: 'carbon.hostedgraphite.com', port: 2003)
```

Comment:
It seems this class simply doesn't work unless we have a valid `api_key`. Should we be explicit and raise when we don't have this attribute?